### PR TITLE
Transaction pipeline

### PR DESCRIPTION
**Pull request by triump2020**
## What type of PR is this?

- [ ] API-change
- [ ] BUG
- [ ] Improvement
- [ ] Documentation
- [x] Feature
- [ ] Test and CI
- [ ] Code Refactoring

## Which issue(s) this PR fixes:

issue #4694 

## What this PR does / why we need it:

### CODE REVIEW

File: pkg/vm/engine/tae/txn/txnbase/txnmgr.go [hunk 1]
```diff
@@ -129,39 +135,65 @@ func (mgr *TxnManager) GetTxn(id uint64) txnif.AsyncTxn {
...
+		mgr.OnException(err)
+	}
+}
+
 func (mgr *TxnManager) onPreparRollback(txn txnif.AsyncTxn) {
 	_ = txn.PrepareRollback()
 }
 
-// TODO
 func (mgr *TxnManager) onPreparing(items ...any) {
```
Comment (XuPeng-SH) [inline pkg/vm/engine/tae/txn/txnbase/txnmgr.go:190]: Could we separate 1PC and 2PC logic?

**Review by XuPeng-SH — CHANGES_REQUESTED** [review]

File: pkg/vm/engine/tae/txn/txnbase/txnmgr.go [hunk 2]
```diff
@@ -129,39 +135,65 @@ func (mgr *TxnManager) GetTxn(id uint64) txnif.AsyncTxn {
 	return mgr.IDMap[id]
 }
 
+func (mgr *TxnManager) EnqueueFlushing(op any) (err error) {
```
[the remark this answers is not in the document]
Comment (triump2020) [reply pkg/vm/engine/tae/txn/txnbase/txnmgr.go:138]: ok

File: pkg/vm/engine/tae/txn/txnbase/txnmgr.go (hunk 1 above)
Comment (triump2020) [reply pkg/vm/engine/tae/txn/txnbase/txnmgr.go:190]: 1PC  and 2PC  transactions must both enter into `onPreparing `for conflict check, etc.   2PC  had been separated from 1PC  inside the `onPreparing `

Comment (triump2020) [reply pkg/vm/engine/tae/txn/txnbase/txnmgr.go:190]: `onPreparing`  means that  do preparing work for  1PC  Commit or  2PC  Prepare

File: pkg/vm/engine/tae/txn/txnbase/txn.go [hunk 3]
```diff
@@ -85,56 +97,176 @@ func (txn *Txn) SetPrepareRollbackFn(fn func(txnif.AsyncTxn) error) { txn.Prepar
...
-	txn.Mgr.DeleteTxn(txn.GetID())
-	err = txn.Err
+	txn.Status = txnif.TxnStatusCommittingFinished
+	return txn.Err
+}
+
+// Commit Notice that the commit of a 2PC transaction must be
+// success once the commit message arrives.
+func (txn *Txn) Commit() (err error) {
+	if (!txn.Is2PC && txn.Status != txnif.TxnStatusActive) ||
```
[the remark this answers is not in the document]
Comment (triump2020) [reply pkg/vm/engine/tae/txn/txnbase/txn.go:204]: NP

**Comment by triump2020** [comment]
@matrix-meow rerun failed

**Comment by triump2020** [comment]
@matrix-meow rerun failed

**Pull request closed without merging**